Rewrite contributing (#2614)
Shopify/ruby-lsp: CONTRIBUTING.md

# CONTRIBUTING

> The Ruby LSP repository has three sub projects inside:

Reviewer: ```suggestion
This repository contains three sub-projects:
```

> - The `ruby-lsp` gem, which is the language server implementation. This is the top level of the repository. Most features are implemented here since everything implemented in the server is available to all editors
> - The VS Code extension, which exists under the `vscode` directory. Any custom VS Code features are implemented here
> - Its documentation website, which exists under the `jekyll` directory. All user facing documentation for both the Ruby LSP and the Rails addon is contained here

Reviewer: ```suggestion
- the **language server** (`ruby-lsp`), which exists at the top level of the repository. Most features are implemented here since everything implemented in the server is available to all editors
- the **VS Code extension**, which exists under the `vscode` directory. Any custom VS Code features are implemented here
- the **documentation** website, which exists under the `jekyll` directory. All user facing documentation for both the Ruby LSP and the Rails addon is contained here
```

This contributing guide is split by each component.

## Initial setup

To start contributing to the Ruby LSP, ensure that all dependencies are installed as follows:

- `bundle install` on the top level will install Ruby dependencies
- `bundle install` in the `jekyll` directory will install documentation dependencies

> - `yarn install` in the `vscode` directory will install NodeJS dependencies for the VS Code extension

Reviewer: ```suggestion
- `yarn install` in the `vscode` directory will install Node.js dependencies for the VS Code extension
```
